### PR TITLE
Changed classes from ES5 to ES6

### PR DESCRIPTION
**Pull request by bekorn**

### CODE REVIEW

File: player.js [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
+        this.rectangle = new Rectangle( center, width, height, this.program );
     }
 
+    draw() {
+        this.rectangle.draw( this.x, this.y );
+    }
 
-    return player;
+    update() {
+        this.x += this.speed;
```
Comment (baranusta) [inline player.js]: update speed first?

File: rectangle.js [hunk 2]
```diff
@@ -1,58 +1,37 @@
...
+        this.program = program;
 
-    rectangle.initialize = function(center, width, height){
-        vao = gl.createBuffer();
-        gl.bindBuffer(gl.ARRAY_BUFFER, vao);
-        var vertices = [
+        this.vao = gl.createBuffer();
+        gl.bindBuffer( gl.ARRAY_BUFFER, this.vao );
+        
+        this.vertices = [
```
Comment (baranusta) [inline rectangle.js]: this could be just a local variable

File: rectangle.js [hunk 3]
```diff
@@ -1,58 +1,37 @@
...
 
-//     gl.bindBuffer(gl.ARRAY_BUFFER, vBuffer);
-//     gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, ebo);
-//     gl.vertexAttribPointer(vPosition, 2, gl.FLOAT, false, 0, 0);
-//     gl.enableVertexAttribArray(vPosition);
+        gl.uniform1f( this.program.x, x );
+        gl.uniform1f( this.program.y, y );
 
-//     gl.drawElements( gl.TRIANGLES, 6, gl.UNSIGNED_SHORT,0);
-//     console.log(gl.getError() == gl.NO_ERROR);
```
Comment (baranusta) [inline rectangle.js]: Do not remove those commented out lines pls

File: rectangle.js [hunk 4]
```diff
@@ -1,58 +1,37 @@
-var Rectangle = function(p){
-    var rectangle={};
-    var program = p;
-    var vao;
-    var vaoAttrib;
```
Comment (baranusta) [inline rectangle.js:5]: those variables are mostly private variables so you should not attach those variables to "this" which makes them accessible from outside of the class(you should check whether it is true).

File: pong_game.js [hunk 5]
```diff
@@ -1,8 +1,8 @@
```
Comment (baranusta) [inline pong_game.js:1]: this file's changes are mostly redundant for the scope of current state of the project

**Review by baranusta — CHANGES_REQUESTED** [review]

File: rectangle.js (hunk 3 above)
Comment (bekorn) [reply rectangle.js]: I'm not sure about the validity of commented codes with this new class syntax, but will keep them

Comment (baranusta) [reply rectangle.js]: This is what I am going to use on next session

File: rectangle.js (hunk 4 above)
Comment (bekorn) [reply rectangle.js:5]: As I understand, in ES6 implementation of classes there is no such thing as private variables (yet). So there is nothing I can do here
[Stackoverflow Question](https://stackoverflow.com/questions/22528967/es6-class-variable-alternatives)

File: player.js (hunk 1 above)
Comment (bekorn) [reply player.js]: I will remove speed from player. Should I also remove the **y** and revert the shader to work with **time**?

**Comment by bekorn** [comment]
Revised the code in a new commit

**Review by baranusta — APPROVED** [review]

**Pull request merged**